STEER/STEERBase/AliInputEventHandler.cxx

```diff
@@ -69,6 +72,40 @@ AliInputEventHandler::AliInputEventHandler(const char* name, const char* title):
 // Named constructor.
 }
 
+//______________________________________________________________________________
+Long64_t AliInputEventHandler::GetEntries()
+{
+  /// Calls fTree->GetEntries and checks the file open status
+  /// Throws AliFatal if any input file could not be opened.
+  Long64_t entries = fTree->GetEntries();
+  // GetEntries tries to open internally all chain files, including the connected friend files
+  // Since open errors are not reported, we need to check the status in the TChain structure
+  TChain *chain = dynamic_cast<TChain*>(fTree);
+  if (chain) {
+    TIter next(chain->GetListOfFiles());
+    TChainElement *elem;
+    while ((elem = (TChainElement*) next())) {
+      if (elem->GetLoadResult() != 0)
+        AliFatal(Form("File %s has status=%d", elem->GetTitle(), elem->GetLoadResult()));
+    }
+  }
+  if (fTree->GetListOfFriends()) {
+    TIter nextfriend(fTree->GetListOfFriends());
+    TFriendElement* fe;
+    while ((fe = (TFriendElement*) nextfriend())) {
+      TTree* t = fe->GetTree();
+      TChain *chain = dynamic_cast<TChain*>(t);
```

Comment:
This is always a chain and always valid?

Reply:
Indeed, I should add a check like for the main chain above. I'll fix this, thanks.

Reply:
This should be fixed now